src/AutoRest.CSharp/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -75,10 +75,19 @@ public async Task<GeneratedCodeWorkspace> ExecuteAsync(Task<CodeModel> codeModel
 
             foreach (var client in context.Library.Clients)
             {
-                var codeWriter = new CodeWriter();
-                clientWriter.WriteClient(codeWriter, client, context.Configuration);
+                var credentialTypes = context.Configuration.CredentialTypes;
+                if (credentialTypes.Contains("TokenCredential", StringComparer.OrdinalIgnoreCase) ||
```

Comment:
Let's not condition ClientOptions on the credentials. I don't think it's an obvious relationship. Using PublicClients might be a bit better.

Reply:
Ahh, ok. I thought we are using ClientOptions only for constructors with `TokenCredential` and `AzureKeyCredential` but I got it now, user can create an instance of ClientOptions and use it anywhere as per their requirements.